inspirehep/modules/theme/views.py

```diff
@@ -101,3 +116,40 @@ def data():
 @blueprint.route('/ping')
 def ping():
     return 'OK'
+
+
+#
+# Feedback
+#
+
+@blueprint.route('/postfeedback', methods=['POST', ])
+def postfeedback():
+    """Handler to create a ticket from user feedback."""
+    feedback = request.form.get('feedback')
+    if feedback == '':
+        return jsonify(success=False), 400
+
+    replytoaddr = request.form.get('replytoaddr', None)
+    if replytoaddr is None:
+        if current_user.is_anonymous:
+            return jsonify(success=False), 403
+        else:
+            replytoaddr = current_user.get('email')
+            if replytoaddr == '':
+                return jsonify(success=False), 403
+
+    content = 'Feedback:\n{feedback}'.format(feedback=feedback)
+    message = {
+        'sender': current_app.config['CFG_SITE_SUPPORT_EMAIL'],
+        'recipients': [current_app.config['INSPIRELABS_FEEDBACK_EMAIL']],
+        'subject': 'INSPIRE Labs Feedback',
+        'body': content,
+        'reply_to': replytoaddr
+    }
+
+    sending = send_email.delay(message)
+
+    if sending.failed():
```

Comment:
Its code is pretty straightforward: http://docs.celeryproject.org/en/latest/_modules/celery/result.html#AsyncResult.failed. Who sets `self.state` to `states.FAILURE`? Celery currently does it in its internals: http://docs.celeryproject.org/en/latest/_modules/celery/backends/base.html#BaseBackend.mark_as_failure. This function is eventually called in the `on_error_callback` of a `Request`.

So, with this we are only catching errors such as "RabbitMQ is down", but not errors in sending the email. On the other hand, if we wait for confirmation of the successful sending of the email then we hang up the UI for a potentially unbounded amount of time...